proxy-socket-core/src/main/java/net/airvantage/proxysocket/core/v2/Tlv.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * MIT License
+ * Copyright (c) 2025 Semtech
+ */
+package net.airvantage.proxysocket.core.v2;
+
+import java.util.Arrays;
+
+public final class Tlv {
+    private final int type;
+    private final byte[] value;
+
+    public Tlv(int type, byte[] data, int offset, int length) {
+        this.type = type;
+        this.value = Arrays.copyOfRange(data, offset, offset + length);
+    }
```

Comment:
strange constructor ? :thinking: 
Why not 
```suggestion
    public Tlv(int type, byte[] data) {
        this.type = type;
        this.value =data; // or eventually data.clone()
    }
```

Reply:
If i update the constructor like you propose i need to put the `Arrays.copyOfRange` part in the `ProxyProtocolV2Decoder.parse` part leading to code duplication